### ISSUE TITLE
Works with lua 5.4 but incorrectly shows the Lua version number on startup

### ISSUE THREAD

**Issue by berwynhoyt** [giann/croissant#12]
When I run croissant it displays Lua 5.3, but when I ask Lua what version is actually running, it says Lua 5.4:
```
$ sudo luarocks --lua-version 5.4 install croissant
...
$ croissant
🥐  Croissant 0.0.1 (C) 2019 Benoit Giannangeli
Lua 5.3 Copyright (C) 1994-2018 Lua.org, PUC-Rio
→ _VERSION
"Lua 5.4"	
→ 
```

**Comment by giann** [comment]
Yes, croissant uses a [trick](https://github.com/giann/croissant/blob/master/croissant/do.lua#L322-L337) to find out the flavor of lua being run. It has not been updated for Lua 5.4 yet.

**Comment by mystix** [comment]
Made a pull request for this:
https://github.com/giann/croissant/pull/13

**Comment by berwynhoyt** [comment]
Why does croissant use these tricks to determine version? Wouldn't it be better to simply check the _VERSION global?

**Comment by mystix** [comment]
Probably because `_VERSION` is just a (global) variable which is easily overwritten?
```
🚀 ❯ croissant                       
🥐  Croissant 0.0.1 (C) 2019 Benoit Giannangeli
Lua 5.3 Copyright (C) 1994-2018 Lua.org, PUC-Rio
→ _VERSION
"Lua 5.4"
→ _VERSION = 'FakeLua 5'
"FakeLua 5"
→ _VERSION
"FakeLua 5"
```

see http://lua-users.org/lists/lua-l/2016-05/msg00297.html for the original code + rationale

**Comment by berwynhoyt** [comment]
True, but if someone overwrites _VERSION then he's probably testing things. If you don't use _VERSION then he can't do the test he wants.

**Comment by mystix** [comment]
I don't know what else luaVersion() is being used for other than to report the true Lua version which `croissant` is running and displaying this value when it starts up.

What I do know is: this hacky-looking code tests for subtleties unique to each Lua version, then uses this to provide a definite Lua version vs having to rely on a variable which might get overwritten anytime.
